apturicovid/api/StatsClient.swift

```diff
@@ -4,71 +4,93 @@ import RxSwift
+    }
+    
+    private func getStatsFromApi(forced: Bool) -> Observable<Stats> {
+        
+        if forced || shouldFetchFromApi() {
+            return Observable<Stats>.create { observer -> Disposable in
+                // Fetch data from api
+                let apiFetch = self.request(urlString: "\(filesBaseUrl)/stats/v1/covid-stats.json", body: nil, method: "GET")
+                    .compactMap{ (data) -> Stats? in
+                        do {
+                            return try Stats.decoder.decode(Stats.self, from: data)
+                        } catch {
+                            observer.onError(error)
+                            return nil
+                        }
+                    }.subscribe(onNext: { stats in
+                        LocalStore.shared.stats = stats
+                        LocalStore.shared.lastStatsFetchTime = Date()
+                        observer.onNext(stats)
+                    }, onError: {
+                        observer.onError($0)
+                    }, onCompleted: {
+                        observer.onCompleted()
+                    })
 
-            let throttle: TimeInterval = {
-                switch Reachability.shared?.connection {
-                case .cellular: return StatsClient.fetchThrottleCellular
-                default:        return StatsClient.fetchThrottleWifi
+                return Disposables.create {
+                    apiFetch.dispose()
                 }
-            }()
-            
-            // Complete if recently got new stats && !forced
-            guard forceFromApi || LocalStore.shared.lastStatsFetchTime.distance(to: Date()) > throttle else {
-                observer.onCompleted()
-                return disposable
             }
-            
-            // Fetch data from api
-            let apiFetch = self.request(urlString: "\(filesBaseUrl)/stats/v1/covid-stats.json", body: nil, method: "GET")
-                .compactMap{ (data) -> Stats? in
-                    do {
-                        return try Stats.decoder.decode(Stats.self, from: data)
-                    } catch {
-                        observer.onError(error)
-                        return nil
-                    }
-                }.subscribe(onNext: { stats in
-                    LocalStore.shared.stats = stats
-                    LocalStore.shared.lastStatsFetchTime = Date()
-                    observer.onNext(stats)
-                    observer.onCompleted()
-                }, onError: {
-                    observer.onError($0)
-                }, onCompleted: {
-                    observer.onCompleted()
-                })
-
-            return Disposables.create {
-                apiFetch.dispose()
+        } else {
+            return Observable.empty()
+        }
+    }
+    
+    private func getStatsFromLocalStorage() -> Observable<Stats> {
+        
```

Comment:
I'd write 

```
`LocalStore.shared.stats.map(Observable.just) ?? Observable.empty()` :)
```

but that's very, very optional.

Reply:
Looks much more compact. Will use your sample